JCL-371: integration module, jsonb dependency

NSeydoux opened the pull request:
This enables running the integration tests with the JSON-B module implementing the JSON service.

NSeydoux committed:
Make mapping class public

JSONB complains otherwise.

NSeydoux committed:
Add bigint support for status credential parsing

In some cases, JSONB instantiates integers as BigInt instances, which wasn't accounted for in the cast to integer.

NSeydoux committed:
Remove inner verification status class

Instead of having a mutable verification status class (which was intended to be private, but JSONB complained about that), the public AccessCredentialVerification class has now public setters and a default constructor to be usable directly by the JSON binding service. The setters will make the attributes immutable, so that they can only be set once.

NSeydoux committed:
Re-enable failing tests

NSeydoux committed:
Implement appropriate JSONB naming strategy

Jsonb current naming strategy fails to correctly parse UMA metadata, or any snake case JSON key that should map to camelcased class attributes starting with a lowercase. This new naming strategy resolves the issue.

langsamu commented on jsonb/src/main/java/com/inrupt/client/jsonb/JsonbService.java:
```diff
@@ -33,20 +33,28 @@
     public JsonbService() {
         final JsonbConfig config = new JsonbConfig();
-        config.withPropertyNamingStrategy(PropertyNamingStrategy.LOWER_CASE_WITH_DASHES);
+        config.withPropertyNamingStrategy(new LowerCamelCase());
```
This is a very general change.
_All_ JSON-B functionality from now on willuse this new naming strategy.

I have no problem with it, except to note that we'll want to communicate in release notes.

langsamu replied:
On second thought, I do see a problem.

JSON-B has its default naming strategy (whatever that is).
Now we make our JSON-B facilities use a different one from the default.
And it's actually one we've written ourselves.
This might confuse developers.

At minimum I'd like to see this prominently reflected in javadocs.
Possibly I'd like us to reconsider this.

Perhaps the use of this naming strategy belongs on a level different from _the_ JSON-B class?

langsamu commented on jsonb/src/main/java/com/inrupt/client/jsonb/JsonbService.java:
```diff
@@ -76,5 +84,28 @@ public <T> T fromJson(final InputStream input, final Type type) throws IOExcepti
         }
     }
+
+    static class LowerCamelCase implements PropertyNamingStrategy {
```
Do we not have a 'standard' implementation of this naming strategy?

By standard I mean one that we don't have to implement ourselves.

langsamu reviewed (changes requested):
I know I myself suggested the use of a naming strategy for controlling JSON serialization.

However see some comments about this particular implementation.

NSeydoux committed:
Revert "Implement appropriate JSONB naming strategy"

This reverts commit 0000bd3af7b931b8e2be906215661cbd837bcd56.

NSeydoux committed:
Implement UMA discovery doc adapter for JSON-B

Instead of changing the global naming strategy, which has side-effects beyond what is reasonable, the JSON-B issue with naming of UMA properties is resolved by implementing an adapter pattern.

NSeydoux committed:
Make snake case json-b default naming strategy

Aligns with Jackson default strategy.

NSeydoux committed:
Update the UMA module to use UmaMetadata from api

It is not possible to remove the Metadata class internal to the uma module entirely from the API, because that would be a breaking change for UmaClient::metadata (a TODO has been added to this effect). Other than that, this enforces that the UMA module codes to the UmaMetadata interface rather than the Metadata class.

NSeydoux committed:
Deserialize UmaMetadata with Jackson

This adds an adapter pattern to deserialize UMA Discovery Documents using Jackson, based on the UmaMetadata interface exposed by the api module.

NSeydoux committed:
fixup! Implement UMA discovery doc adapter for JSON-B

NSeydoux committed:
fixup! Update the UMA module to use UmaMetadata from api

NSeydoux committed:
fixup! Implement UMA discovery doc adapter for JSON-B
